lib/mutations/uri_filter.rb

```diff
@@ -0,0 +1,45 @@
+require 'uri'
+
+module Mutations
+  class UriFilter < AdditionalFilter
+    @default_options = {
+      :nils => false,    # true allows an explicit nil to be valid. Overrides any other options
+      :scheme => nil,    # restrict the URI to a specific scheme, i.e. 'https'
+    }
+
+    def filter(data)
+      if data.nil?
+        return [nil, nil] if options[:nils]
+        return [nil, :nils]
+      end
+
+      case data
+      when URI # we're good!
+      when String then
+        return [nil, :blank] if blank?(data)
+        begin
+          data = URI.parse(data)
+        rescue StandardError => e
+          return [nil, e.message]
```

Comment:
@estraph The filter operation always starts from the `HashFilter` and from there it will filter the nested input. The results of the nested filter will always flow back to `HashFilter`. 

What I can see is all basic filter like `StringFilter`, `FloatFilter` and etc always return `Array<Object, Symbol>` in `#filter`. Are you sure returning `e.message` here will be handled gracefully by the error handler?

I think you might see the issue if you write a rspec with a `Mutation::Commands` that will trigger error on this filter.